metagraph/core/planning.py

```diff
@@ -144,15 +163,14 @@ def __init__(
         resolver,
         concrete_algorithm,
         required_translations: Dict[str, MultiStepTranslator],
+        build_problem_messages: List[str],
```

Comment:
I understand the purpose of `build_problem_messages`, but let's rename it to `err_msgs` and also have an `unsatisfiable` boolean variable. That will make code using the AlgorithmPlan much easier to read. `if plan.unsatisfiable:` vs `if plan.build_problem_messages == 0:`.

Reply:
https://github.com/metagraph-dev/metagraph/pull/110/commits/963fa1038c4ce5c505368484e8ea4a93d6605601 intends to address this problem. 

Rather than adding a separate boolean variable named `unsatisfiable`, I added the method `AlgorithmPlan. unsatisfiable` via the `@property` decorator so that we don't have to worry about potential bugs where `plan.unsatisfiable` might differ from `plan.build_problem_messages == 0`. 

@jim22k , does this sound like a reasonable approach?

Reply:
Yes, that is a good approach.